src/dynamic_boundary_conditions/sea_level_rise_data.py

```diff
@@ -3,106 +3,95 @@
+    gdf = gpd.GeoDataFrame(df, crs=crs, geometry=gpd.points_from_xy(df.lon, df.lat))
+    gdf.rename(columns={'siteId': 'Site_ID', 'year': 'Projected_Year',
+                        'p17': 'SLR_p17_confidence_interval_meters',
+                        'p50': 'SLR_p50_confidence_interval_meters',
+                        'p83': 'SLR_p83_confidence_interval_meters',
+                        'lon': 'Longitude', 'lat': 'Latitude',
+                        'measurementName': 'Scenario_(VLM_with_confidence)', },
+               inplace=True)
+    return gdf
+
+
+def save_to_database(connect, gdf: gpd.GeoDataFrame, table_name: str, if_exists: str = 'replace'):
+    """ save dataframe to database """
+    gdf.to_postgis(table_name, connect, index=False, if_exists=if_exists)
 
 
-def closest_site_location(locations_lat_lon: list, user_value: dict):
-    """"Returns the haversine closest SLR site location"""
-    return min(locations_lat_lon, key=lambda p: haversine((user_value['lat'], user_value['lon']), (p['lat'], p['lon'])))
-
-
-def nearest_site(locations_lat_lon: list, lat_input: float, lon_input: float):
-    """ Returns lat lon coordinates which is then used by site_data function
-        Selecting only the data for chosen site (e.g. 4303 is site just south
-        of Waimakiriri River Mouth which is closest to a selected location of
-        ('lat': -43.391266, 'lon': 172.715979)"""
-
-    user_value = {'lat': lat_input, 'lon': lon_input}
-    site = closest_site_location(locations_lat_lon, user_value)
-    print('Closest Site Location: ', str(site.get("lat")) + "," + str(site.get("lon")))
-    print('Searching for site data: ')
-    return Point(site.get("lon"), site.get("lat"))
-
-
-def site_data(sea_level_rise_projections_nz: gpd.geodataframe.GeoDataFrame, locations_lat_lon: list, lat_input: float,
-              lon_input: float):
-    """ Returns data for closest site by querying user for their desired location (calls nearest_site function) """
-    if lat_input < -90 or lat_input > 90:
-        raise ValueError("Latitude is out of range [-90, 90]")
-    if lon_input < -180 or lon_input > 180:
-        raise ValueError("Longitude is out of range [-180, 180]")
-
-    return sea_level_rise_projections_nz.loc[
-        sea_level_rise_projections_nz['geometry'] == nearest_site(locations_lat_lon, lat_input, lon_input)]
+def main():
 
+    engine = setup_environment.get_database()
 
-def main():
-    path = "data/"
+    path = "./src/dynamic_boundary_conditions/data/"
 
-    ## Fetching centroid co-ordinate from user selected shapely polygon
+    # Fetching centroid co-ordinate from user selected shapely polygon
     lat = Polygon([[-43.298137, 172.568351], [-43.279144, 172.833569], [-43.418953, 172.826698],
                    [-43.407542, 172.536636]]).centroid.coords[0][0]
     long = Polygon([[-43.298137, 172.568351], [-43.279144, 172.833569], [-43.418953, 172.826698],
                     [-43.407542, 172.536636]]).centroid.coords[0][1]
```

Comment:
This polygon should be defined once in a variable, then you can retrieve the lat and long from it.
ie:
```
example_polygon_centroid = Polygon([[-43.298137, 172.568351], [-43.279144, 172.833569],
                                   [-43.418953, 172.826698], [-43.407542, 172.536636]]).centroid
lat, long = example_polygon_centroid.coords[0]
```

Reply:
Good practice, add to the next commit.